tests/integration.rs

```diff
@@ -30,31 +23,50 @@ fn int1() {
         };
         let exec_config = pewpew::ExecConfig::Run(run_config);
 
-        let stdout = test_common::TestWriter::new();
-        let stderr = test_common::TestWriter::new();
+        let stdout = TestWriter::new();
+        let stderr = TestWriter::new();
 
+        let stdout2 = stdout.clone();
         let stderr2 = stderr.clone();
 
-        let did_succeed = Arc::new(AtomicBool::new(false));
```

Comment:
We don't need any of the async stuff for this test anymore? Was it trying to run this test in parallel with others?

Reply:
It's still async, but the async pieces moved to the `run_test` function. Now, because we use the `.block_on` method on the Tokio `Runtime` the integration tests can segment the async stuff behind a blocking call. Essentially saying "runtime, you handle all that async stuff and let me know when everything is done".